### PR TITLE
Use JSON.parse on query string values in case they are arrays or objects

### PR DESCRIPTION
**Pull request by Vijar**
Also, use supertest for better coverage.

@mridgway @redonkulus

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

File: libs/fetcher.js [hunk 1]
```diff
@@ -6,18 +6,34 @@
...
+var OP_UPDATE = 'update';
+var GET = 'GET';
+var qs = require('querystring');
+var debug = require('debug')('Fetchr');
+
+function parseValue(value) {
+    // take care of value of type: array, object
+    try {
+        return JSON.parse(value);
+    } catch (ignore) {
```
Comment (jedireza) [inline libs/fetcher.js]: Rename `ignore` to `e`, `error` or `exception`?

**Comment by jedireza** [comment]
:+1:

File: package.json [hunk 2]
```diff
@@ -27,13 +27,16 @@
     "xhr": "^2.0.0"
   },
   "devDependencies": {
+    "body-parser": "~1.12.2",
```
Comment (redonkulus) [inline package.json]: ^

File: package.json [hunk 3]
```diff
@@ -27,13 +27,16 @@
...
     "chai": "^2.0.0",
     "coveralls": "^2.11.1",
+    "express": "~4.12.3",
```
Comment (redonkulus) [inline package.json]: ^

File: libs/fetcher.js (hunk 1 above)
Comment (Vijar) [inline libs/fetcher.js]: JSLint used to expect this to be `ignore` but I think it can be anything now.

**Comment by Vijar** [comment]
@redonkulus @jedireza updated.

File: tests/unit/libs/fetcher.client.js [hunk 4]
```diff
@@ -39,42 +44,29 @@ describe('Client Fetcher', function () {
...
-                    pathname = decodeURIComponent(pathname);
-                    pathname = pathname.split(';');
-                    expect(pathname.shift()).to.equal('/api/resource/' + resource);
-                    while(!!(pair = pathname.shift())) {
-                        var pair = pair.split('=');
-
-                        var k = pair[0],
-                            v = pair[1];
-
-                        //hacky because of array in querystring, but its fine
```
Comment (Vijar) [inline tests/unit/libs/fetcher.client.js]: @redonkulus was the hack to get around arrays not being serialized properly before...

**Comment by redonkulus** [comment]
:shipit:

**Pull request merged**